tasks/eva_3417/src/main/groovy/eva3417/HashCollisionWithDifferentRSRemediation.groovy

```diff
@@ -0,0 +1,573 @@
+        if (sveList != null && !sveList.isEmpty()) {
+            return sveClass
+        } else {
+            List<SubmittedVariantEntity> dbsnpSveList = dbEnv.mongoTemplate.find(query(where("_id").is(hash)), dbsnpSveClass)
+            if (dbsnpSveList != null && !dbsnpSveList.isEmpty()) {
+                return dbsnpSveClass
+            } else {
+                throw new RuntimeException("SVE with hash " + hash + " could not be found")
+            }
+        }
+    }
+
+    List<ClusteredVariantEntity> getCVEInDB(Long... accessions) {
+        Set<Long> accList = new ArrayList<>()
+        for (Long acc : accessions) {
+            accList.add(acc)
+        }
+        logger.info("getting CVE for accessions " + gson.toJson(accList))
+        List<ClusteredVariantEntity> cveInDBList = [cveClass, dbsnpCveClass]
+                .collectMany(cveClass -> dbEnv.mongoTemplate.find(query(where("accession").in(accList)), cveClass))
+                .flatten()
+
+        return cveInDBList
+    }
+
+    boolean checkIfValidRS(Long rs, String asm, List<ClusteredVariantEntity> cveList) {
+        if (rs == null) {
+            return false
+        } else {
+            Optional<ClusteredVariantEntity> opRS = cveList.stream()
+                    .filter(cve -> cve.getAccession() == rs && cve.getAssemblyAccession() == asm)
+                    .findAny()
+            if (opRS.isPresent()) {
+                return true
+            } else {
+                return false
+            }
+        }
+    }
+
+    CATEGORY getCollisionCategory(Long rsInFile, Long rsInDB, boolean rsInFileValid, boolean rsInDBValid) {
+        if (rsInFile == null && rsInDB == null) {
+            // none of the SVE contains any RS
+            return CATEGORY.NO_SVE_HAS_RS
+        } else if (rsInFile != null && rsInDB != null) {
+            // both the SVE contains RS
+            if (!rsInFileValid && !rsInDBValid) {
+                // both the RS are invalid
+                return CATEGORY.BOTH_SVE_HAS_RS_NONE_VALID
+            } else if (rsInFileValid && rsInDBValid) {
+                //both the RS are valid
+                return CATEGORY.BOTH_SVE_HAS_RS_BOTH_VALID
+            } else if (!rsInFileValid || !rsInDBValid) {
+                // one RS is valid while the other is Invalid
+                return CATEGORY.BOTH_SVE_HAS_RS_ONE_VALID
+            }
+        } else if (rsInFile != null || rsInDB != null) {
+            // one of the SVE contains RS
+            if (!rsInFileValid && !rsInDBValid) {
+                // one of the SVE contains RS but that RS is not valid
+                return CATEGORY.ONE_SVE_HAS_RS_NONE_VALID
+            } else if (rsInFileValid || rsInDBValid) {
+                // one of the SVE contains a valid RS
+                return CATEGORY.ONE_SVE_HAS_RS_ONE_VALID
+            }
+        }
+    }
+
+    List<SubmittedVariantEntity> findSVEAccessionInOrgAssembly(String asm, Long acc) {
```

Comment:
```suggestion
    List<SubmittedVariantEntity> findSVEAccessionInOrigAssembly(String asm, Long acc) {
```
I think this is what you mean? I would actually make this a function of the SVE and call `getRemappedFrom()` and `getAccession()` inside the method, saves some code repetition and makes the intent of the method clearer.

Reply:
done